### PR TITLE
Add league scraper

### PR DESCRIPTION
**Pull request by PratyushTrivedi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
-*.pyc
-
```
Comment (ms3001) [inline .gitignore:2]: Can you keep these files here too

File: src/database.py [hunk 2]
```diff
@@ -13,12 +13,12 @@ def __setstate__(self, state):
 
   def AddNewPlayer(self, player):
     if player in self.players:
-      print("Nice try NERD this player already exists!")
+      #print("Nice try NERD this player already exists!")
```
Comment (ms3001) [inline src/database.py:18]: uncomment this and below

File: src/database.py [hunk 3]
```diff
@@ -30,4 +30,18 @@ def SetPlayerElo(self, player, elo):
...
\ No newline at end of file
+      " Games played: " + str(self.players[player].games_played))
+
+  def GetPlayer(self, player):
+    if player in self.players:
+      return self.players[player]
+    else:
+      raise IndexError("Player Not Found")
+
+  def PrintDatabaseSortedElo(self):
```
Comment (ms3001) [inline src/database.py]: suggest `PrintDatabaseSortedByElo`

File: src/driver.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import pickle
-
+from league_scraper_v1 import scrapeMatchHistoryUrl
```
Comment (ms3001) [inline src/driver.py:5]: add a new line here (I wanted to keep a space between external vs internal deps)

File: src/driver.py [hunk 5]
```diff
@@ -41,11 +42,11 @@ def StartDecisionLoop():
     AddPlayer(player)
 
   elif val == 'g':
-    print("Current datbase: ")
-    db.PrintDatabase()
+    print("Current database: ")
+    db.PrintDatabaseSortedElo()
 
   elif val == 's':
```
Comment (ms3001) [inline src/driver.py:52]: should we get rid of this now?

File: src/inputUrls.txt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+https://matchhistory.na.leagueoflegends.com/en/#match-details/NA1/3609773279/48333464?tab=overview
```
Comment (ms3001) [inline src/inputUrls.txt]: why do we want to merge this?

File: src/league_scraper_v1.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+	time.sleep(8) #sleep while page loads
+
+	page_source = browser.page_source
+	soup = BeautifulSoup(page_source, 'html.parser')
+
+	browser.close()
+
+	#test print statements
+	#print(soup.prettify())
```
Comment (ms3001) [inline src/league_scraper_v1.py:46]: suggest removing comments or uncommenting, here and below

File: src/league_scraper_v1.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	username.send_keys(usernametext)
+
+	password = browser.find_element_by_name("password")
+	password.clear();
+	password.send_keys(passwordtext)
+
+	submit = browser.find_element_by_css_selector(".mobile-button__submit")
+	submit.click()
+
+	time.sleep(8) #sleep while page loads
```
Comment (ms3001) [inline src/league_scraper_v1.py]: there should be a better way to do this, but not important for now. can you leave a todo?

File: src/database.py (hunk 2 above)
Comment (PratyushTrivedi) [reply src/database.py:18]: when adding 24 matches it trashes up the output on cmd line

Comment (PratyushTrivedi) [reply src/database.py:18]: every match it adds every player again

Comment (ms3001) [reply src/database.py:18]: ahhh true. let's delete it then

File: src/inputUrls.txt (hunk 6 above)
Comment (PratyushTrivedi) [reply src/inputUrls.txt]: we can keep this running list of urls so anyone can recompile all the matches, which we need to do if we add scraping of more specific match data

File: src/league_scraper_v1.py (hunk 7 above)
Comment (PratyushTrivedi) [reply src/league_scraper_v1.py:46]: leave them for now, will revisit and use these print statements when updating scraper to get more data from each match

File: src/inputUrls.txt (hunk 6 above)
Comment (ms3001) [reply src/inputUrls.txt]: sounds good! can we move it to a different folder? (data/ or similar)

File: src/driver.py [hunk 9]
```diff
@@ -105,37 +116,105 @@ def SetupGame():
...
+      team_1.append(db.GetPlayer(player))
+    for player in result[1][5:]:
+      team_2.append(db.GetPlayer(player))
+
+    game.SetTeam1(team_1)
+    game.SetTeam2(team_2)
+
+    game.FinishGame(int(result[0]))
+
+    game = Game()
```
Comment (ms3001) [inline src/driver.py]: Can we make a function for this? looks identical to code above.

File: src/driver.py [hunk 10]
```diff
@@ -95,6 +105,7 @@ def SetupGame():
     print("Not enough players to start a game! Must have at least 10.")
     return
   
+  pList = []
   team_1 = []
   team_2 = []
```
Comment (ms3001) [inline src/driver.py:111]: move these below line 134

File: src/league_scraper_v1.py (hunk 8 above)
Comment (PratyushTrivedi) [reply src/league_scraper_v1.py]: slighter better version exists now

File: .gitignore (hunk 1 above)
Comment (ms3001) [reply .gitignore:2]: still need to add back the folders below :P

File: src/database.py (hunk 2 above)
Comment (ms3001) [reply src/database.py:18]: this still needs to be deleted

**Review by ms3001 — APPROVED** [review]

**Pull request merged**